### ISSUE TITLE
Here's how we improved our builds

### ISSUE THREAD

**Issue by beefon** [CocoaPods/CocoaPods#9649]
## Preambula

We use CocoaPods (1.9), and we have ~170 devpods. Those are dynamic frameworks which we link as static in Release builds using CocoaPods Amimono-like plugin.

We experienced a huge LLDB slow-on-breakpoint issue.

## Research

After discussing this on [Swift forums](https://forums.swift.org/t/lldb-is-slow-to-resolve-local-vars/32517/58) we end up patching resulting projects and support files after `pod install` finishes the following way:

- Removing most of `modulemap` files generated by CocoaPods
- Excluding `dummy.m` files from compilation process.

## Impact

- This improved LLDB speed by 45%. 
- This also reduced compilation time: e.g. clean build of tests 239 seconds -> 208 seconds.

## Caveats

-  CocoaPods generated modulemap allows to mix ObjC and Swift as well. Also, `modulemap` files have a reference to umbrella header that includes UIKit (for convenience I guess, but that's debatable).
- `dummy.m` allows to build source-less pods I guess. 

## Appeal to the CocoaPods team

- Please consider allowing to disable `modulemap` generation, and allow to provide a replacement natively e.g. using Podfile syntax. This will break compilation in most code bases thought, as `import UIKit` must be added explicitly to the source files. But I think any explicitness is better than implicitness. Optionality of this feature can save most users from fixing their imports.
- Please consider allowing to remove `dummy.m` files - I bet this is where build speed impact comes from.

**Comment by dnkoutso** [comment]
Yes! As you say, the breaking change is a problem so it will have to be for 2.0.

**Comment by beefon** [comment]
This can be done as optional functionality, and can be enabled in podfile somehow as a feature flag, e.g. like `generate_multiple_pod_projects` but on per-pod basis.

**Comment by dnkoutso** [comment]
I havent read the Swift forums thread in detail but why does: 

> Excluding dummy.m files from compilation process.

help so much with this?

**Comment by beefon** [comment]
I guess because Xcode doesn't have to execute clang. Even though the dummy class is simple, clang execution takes some time.
